### ISSUE TITLE
[elastic-7] 2nd iteration

### ISSUE THREAD

**Issue by jesusgn90** [wazuh/wazuh-kibana-app#1433]
#### Define a new naming convention for Elastic stack configuration files

> Wazuh 3.9 is compatible with both Elastic 6.x and Elastic 7.x as well so we must think directory tree for our configuration files, also give them a proper name.

Proposal: 

- Remove `-7` suffix, move 6.x files to `extensions/6.x`
- Upload all the configuration files to S3, out of the wazuh/wazuh repository.

#### Remove Logstash as a required component

> Wazuh is using Logstash mainly for geolocation fields and it also removes some other fields. We are also dropping events that have no `rule` field as a workaround for Beats related events.

Proposal: 

- Enforce Filebeat usage as a required component and modify its related configuration, this includes load balancer, mutating fields, ensure all events are being inserted properly into Elasticsearch.

#### Remove Java as a required component

> Since Elastic 7.x, Java is no longer needed for Elasticsearch but Logstash. Wazuh is using Java for Logstash, if there is no Logstash, we don't need Java.

Proposal:

- Remove it from the documentation.

#### Simplified documentation

> With the above modifications, we'd simplify a lot our documentation, having fewer steps and a lower error margin for any kind of installation.

Proposal:

- Review all the documentation where the Elastic stack is involved to fit the new proposals.

#### Review app design for Kibana 7

> Currently, the app is already fixed to work properly with Kibana 7 but its design (mainly the color palette) is a bit weird because Kibana 7 changed all the theme.

Proposal:

- Create a new theme from scratch, compatible with the dark mode too. Using all the React components from the EUI framework.

#### Review alerts template

> We have tons of "keyword" fields, we must review them, probably all of them are not needed. Also, with the above comments, the template might suffer some other changes, further details are needed.

Proposal:

- Reduce the alerts template as much as possible.

**Comment by juankaromo** [comment]
This is an updated branch with some dark theme adaption using eui components
https://github.com/wazuh/wazuh-kibana-app/tree/app-dark-theme

![image](https://user-images.githubusercontent.com/21195730/57357246-c6fa1780-7172-11e9-95e2-7c586dbfca94.png)

**Comment by jesusgn90** [comment]
## Update 08/05/2019

**No more Logstash, no more Java**

After some researching I've concluded Logstash is no longer needed. It implies that Java is no longer required in a default installation. Using Filebeat with some tweaks we can still have the same results in Elasticsearch, also you can still use Logstash if needed, replacing some settings in the Filebeat configuration, it's not a problem.

**New Filebeat configuration file**

For the raw content please see https://github.com/wazuh/wazuh/blob/elastic-7-2/extensions/filebeat/filebeat.yml, here I'm going to explain every part of the file.

The input section refers to the source log file to be read. In our use case, it's the `alerts.json` with `log` as type:

```yaml
filebeat.inputs:
  - type: log
    paths:
      - '/var/ossec/logs/alerts/alerts.json'
```

The template is now being inserted (if it's missing) into Elasticsearch with the next four lines, this is a good improvement, now the log shipper will take care about the template (read the documentation changes in the next points of this comment):

> Template from a JSON file, located at "/etc/filebeat/wazuh-template.json", the name to be stored in Elasticsearch is "wazuh" and it overwrites any other template named "wazuh"

```yaml
setup.template.json.enabled: true
setup.template.json.path: "/etc/filebeat/wazuh-template.json"
setup.template.json.name: "wazuh"
setup.template.overwrite: true
```

This processor parses the event and allow us to enrich/modify/alter the event so it fits our needs (more information https://www.elastic.co/guide/en/beats/filebeat/current/decode-json-fields.html):

```yaml
processors:
  - decode_json_fields:
      fields: ['message']
      process_array: true
      max_depth: 200
      target: ''
      overwrite_keys: true
```

Here we are removing fields from the event before being sent to Elasticsearch. Most of them are extra fields added by Filebeat. Here we have **one stopper**, the field `timestamp` from our alerts is being dropped temporary, Filebeat adds its own @timestamp field and there is no way to replace it with our `timestamp` field. Tomorrow we will find the solution for this, it will be probably to use `timestamp` as the reference field for time-based indices, it's also needed to modify some modules of the Wazuh app where we are using @timestamp such as the index patterns or some queries to Elasticsearch.

```yaml
  - drop_fields:
      fields: ['message', 'ecs', 'beat', 'input_type', 'tags', 'count', '@version', 'log', 'offset', 'type', 'host', 'timestamp']
```
The renaming section, this is very similar to https://github.com/wazuh/wazuh/blob/18af17e5afca7c15e23dc554f4cd75f1ab37bdc1/extensions/logstash/6.x/01-wazuh-remote.conf#L13-L29, but some details differ from the original Logstash filter with some changes.

- If the incoming event has "data.aws.sourceIPAddress", "data.win.eventdata.ipAddress", "data.srcip" and it matches the regexp `\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b`, then copy its value to "@src_ip" and remove the original key. This is intended because we want that field for a possible geolocation later in Elasticsearch. 
- If none of those three fields is present, then ignore these procedures (`ignore_missing: true`). 
- Never crash the process (`fail_on_error: false`)


```yaml
  - rename:
      fields:
        - from: "data.aws.sourceIPAddress"
          to: "@src_ip"
      ignore_missing: true
      fail_on_error: false
      when:
        regexp:
          data.aws.sourceIPAddress: \b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b
  - rename:
      fields:
        - from: "data.srcip"
          to: "@src_ip"
      ignore_missing: true
      fail_on_error: false
      when:
        regexp:
          data.srcip: \b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b
  - rename:
      fields:
        - from: "data.win.eventdata.ipAddress"
          to: "@src_ip"
      ignore_missing: true
      fail_on_error: false
      when:
        regexp:
          data.win.eventdata.ipAddress: \b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b
```

Finally, here we have the output, where the user adds the Elasticsearch nodes where the alerts will be sent.

- On a side note, you can see `#pipeline: geoip`, it's related to the geolocation features, here you can see how to enable it https://github.com/wazuh/wazuh-documentation/blob/elastic-7/source/installation-guide/installing-elastic-stack/geoip.rst, in any case, I'm going to explain it in this comment too.

```yaml
output.elasticsearch:
  hosts: ['http://YOUR_ELASTIC_SERVER_IP:9200']
  #pipeline: geoip
  indices:
    - index: 'wazuh-alerts-3.x-%{+yyyy.MM.dd}'
```

At this point, we've explained Filebeat/Logstash related changes, now it's time to talk about the new documentation. For details, you can see the PR https://github.com/wazuh/wazuh-documentation/pull/1150, here I'm going to explain the most relevant points (by now, the PR has  32 commits 32 and 38 files changed).

**Enrich certain events with geolocation**

As we explain here https://github.com/wazuh/wazuh-documentation/blob/elastic-7/source/installation-guide/installing-elastic-stack/geoip.rst, it's disabled by default, but it's easy to enable it.

Create a new ingest pipeline for `geoip` in Elasticsearch, just one curl:

```js
curl -X PUT "localhost:9200/_ingest/pipeline/geoip" -H 'Content-Type: application/json' -d'
{
    "description" : "Add geoip info",
    "processors" : [
        {
        "geoip" : {
            "field" : "@src_ip",
            "target_field": "GeoLocation",
            "properties": ["city_name", "country_name", "region_name", "location"],
            "ignore_missing": true
        }
        }
    ]
}
'
```

Replace `#pipeline: geoip` with `pipeline: geoip` in `/etc/filebeat/filebeat.yml`, then restart Filebeat, that's all.

**Removed documentation about Logstash and Java**

This is one of the main improvements here, it simplifies the documentation reducing the error margin and the number of steps needed for any deployment, it also reduces the amount of RAM/CPU an instance needs because Logstash is heavy and it uses Java, Filebeat is much cheaper.

**Template step is now safer and it "doesn't need" Elasticsearch** 

As we can see here https://github.com/wazuh/wazuh-documentation/blob/elastic-7/source/installation-guide/installing-wazuh-server/wazuh_server_rpm.rst#installing-filebeat, now the user downloads the template at Filebeat installation time, and downloads it in `/etc/filebeat`.

`Now, Filebeat will insert the template if it's missing and not the user.`

1. Install Filebeat
2. Curl the template in `/etc/filebeat`
3. Curl the Filebeat configuration in `/etc/filebeat`
4. Start the service.

And Elasticsearch is now:

1. Install Elasticsearch
2. Start the service

Now, **it's no longer needed to wait until Elasticsearch is ready**, also, it's no longer needed to take care of the template insertion, because Filebeat will do it.

**Distributed vs Single host unified**

Now there is no difference between a distributed architecture and a single-host architecture because every Wazuh manager has Filebeat and it points directly to Elasticsearch, and it also inserts the template if it's missing, so the documentation has been changed for it.

**New naming convention for wazuh/wazuh/extensions**

Here is the proposal for the new files:

```
extensions/
├── elasticsearch/
│   ├── restore_alerts/
│   │   ├── README.md
│   │   ├── restore_alerts.conf
│   │   └── restore_alerts.sh
│   ├── wazuh-elastic6-template-alerts.json
│   └── wazuh-elastic7-template-alerts.json
├── filebeat/
│   ├── 6.x/
│   │   └── filebeat.yml
│   └── filebeat.yml
├── logstash/
│   └── 6.x/
│       ├── 01-wazuh-local.conf
│       └── 01-wazuh-remote.conf
```

For Elastic 7, the user needs extension/elasticsearch/wazuh-elastic7-template-alerts.json and extensions/filebeat/filebeat.yml, other files are for 6.x.

**Other docs have been simplified too**

I think I've already told you this, but as a reminder, it's no longer needed the "increasing memory for node" paragraphs in Kibana app, no need to wait until the optimization finishes.

No need to wait until Elasticsearch is ready for the template, even better, no need to have Elasticsearch installed when setting up the template, Filebeat will do it once Elasticsearch is ready.

### Next steps

- Review alerts template fields, reduce if possible, we want to simplify all the components.
- Find a good solution for `timestamp` field.
- Review app design

**Comment by jesusgn90** [comment]
## Update 09/05/2019

**Wazuh app**

- Using `timestamp` instead of `@timestamp` in all places where the app was using `@timestamp` https://github.com/wazuh/wazuh-kibana-app/commit/6fe90164456f0e13f43bd88403e36fb96a5c6fa8
- Exclude `@timestamp` from the index pattern in Kibana (hide it from documents), `sourceFilters: '[{"value":"@timestamp"}]'` is the trick here (included in https://github.com/wazuh/wazuh-kibana-app/commit/6fe90164456f0e13f43bd88403e36fb96a5c6fa8 too)
- Minor changes not related directly to Kibana 7 https://github.com/wazuh/wazuh-kibana-app/commit/23b85a0c19dd078a1b67193044663a828d234e85, https://github.com/wazuh/wazuh-kibana-app/commit/07f1e841d1e1470d72923d5c432ccf49812abe3b

**Configuration files**

The only change made was https://github.com/wazuh/wazuh/commit/a30aad3c397f0a84d6d5356cbe68b34bc028a2ab, where I restored the `timestamp` field and add it the date format `"date_optional_time||epoch_millis"` which can parse our timestamp date format. However, I've noticed some alerts are not giving the right padding for the milliseconds part, this is a @wazuh/core task, all the timestamps should have the next format but we can see not all the alerts respect that format:

```
2019-05-09T14:03:05.6+0000
2019-05-09T14:04:04.36+0000
2019-05-09T16:08:58.462+0000
```

The modification I did to the template works fine, but I've opened a ticket https://github.com/wazuh/wazuh/issues/3293 so we can use the right format with no workarounds.

**Some Filebeat testing**

- Filebeat started before downloading the alerts template (remember that now Filebeat handles the template) -> It doesn't matter because Filebeat will use `filebeat-*` indices and not `wazuh-alerts*` indices as Logstash does, so no more corrupted indices due to a missing template. The worst case is having no index for today but the `filebeat*` index, just reindex it.
- Filebeat started before downloading the alerts template, then I restarted Filebeat. In this case, Filebeat switch the destination index, now it will use the `wazuh-alerts*` after sending the template.
- Elasticsearch is down, it doesn't matter, Filebeat has the registry pointer, once Elasticsearch is ready again, Filebeat resumes the job, this is not new, it's the standard behavior.

From my view, the **template-related errors** are **reduced** by **85%**.

**Documentation**

- Fix typo in a diagram: https://github.com/wazuh/wazuh-documentation/commit/554be6fc3310d40995854ba3327feafe25f39ce8

**Comment by jesusgn90** [comment]
## Update 10/05/2019

Today I've been testing SSL communication through the different components of the Elastic stack.

Here is a simple diagram of the architecture:

![image](https://user-images.githubusercontent.com/8860227/57540541-3cb9db00-734d-11e9-965e-1e613360278d.png)

1) Adding simple auth for Elasticsearch:

> Node 01

```sh
# Add trial license
curl -X POST "172.16.1.2:9200/_xpack/license/start_trial?acknowledge=true"

# /etc/elasticsearch/elasticsearch.yml
xpack.security.enabled: true

# Generate credentials
systemctl restart elasticsearch
/usr/share/elasticsearch/bin/elasticsearch-setup-passwords auto
elastic = random_password
```

> Node 02

```sh
# /etc/elasticsearch/elasticsearch.yml
xpack.security.enabled: true

systemctl restart elasticsearch
```

2) Add credentials for both Filebeat and Kibana:

> Node 01

```sh
# /etc/filebeat/filebeat.yml
output.elasticsearch.username: elastic
output.elasticsearch.password: random_password

# /etc/kibana/kibana.yml
elasticsearch.username: elastic
elasticsearch.password: random_password
xpack.security.enabled: true
```

3) Generate CA, certificates, and keys:

> Create `instances.yml` file where we define the instances of our environment:

```yml
instances:
  - name: "es-node-1"
    ip:
      - "172.16.1.2"
  - name: "es-node-2"
    ip:
      - "172.16.1.3"
```

> Generate the certificate bundle zip:

```
# Just once in node 02
./bin/elasticsearch-certutil cert ca --pem --in instances.yml --out certs.zip
```

> Directory tree for certs.zip extracted

```
certs/
|-- ca
|   `-- ca.crt
|-- es-node-1
|   |-- es-node-1.crt
|   `-- es-node-1.key
`-- es-node-2
    |-- es-node-2.crt
    `-- es-node-2.key
```

4) Setting up certificates for all the components:

> Node 01

```sh
mkdir /etc/elasticsearch/certs/ca -p
mkdir /etc/elasticsearch/certs/es-node-01
cp certs/ca/ca.crt /etc/elasticsearch/certs/ca
cp certs/es-node-1.crt /etc/elasticsearch/certs
cp certs/es-node-1.key /etc/elasticsearch/certs
chmod 777 -R /etc/elasticsearch/certs # Excessive, just testing
```

> Node 02

```sh
mkdir /etc/elasticsearch/certs/ca -p
mkdir /etc/elasticsearch/certs/es-node-02
cp certs/ca/ca.crt /etc/elasticsearch/certs/ca
cp certs/es-node-2.crt /etc/elasticsearch/certs
cp certs/es-node-2.key /etc/elasticsearch/certs
chmod 777 -R /etc/elasticsearch/certs # Excessive, just testing
```

> Filebeat (using the same cert as Node 01, same instance)

```sh
mkdir /etc/filebeat/certs/ca -p
mkdir /etc/filebeat/certs/es-node-02
cp certs/ca/ca.crt /etc/filebeat/certs/ca
cp certs/es-node-2.crt /etc/filebeat/certs
cp certs/es-node-2.key /etc/filebeat/certs
chmod 777 -R /etc/filebeat/certs # Excessive, just testing
```

> Kibana (using the same cert as Node 01, same instance)

```sh
mkdir /etc/kibana/certs/ca -p
mkdir /etc/kibana/certs/es-node-02
cp certs/ca/ca.crt /etc/kibana/certs/ca
cp certs/es-node-2.crt /etc/kibana/certs
cp certs/es-node-2.key /etc/kibana/certs
chmod 777 -R /etc/kibana/certs # Excessive, just testing
```

5) SSL settings for Elasticsearch nodes:

> Node 01 (transport layer + http layer)

```yml
# /etc/elasticsearch/elasticsearch.yml 
xpack.security.transport.ssl.enabled: true
xpack.security.transport.ssl.verification_mode: certificate
xpack.security.transport.ssl.key: /etc/elasticsearch/certs/es-node-1/es-node-1.key
xpack.security.transport.ssl.certificate: /etc/elasticsearch/certs/es-node-1/es-node-1.crt
xpack.security.transport.ssl.certificate_authorities: [ "/etc/elasticsearch/certs/ca/ca.crt" ]                

xpack.security.http.ssl.enabled: true
xpack.security.http.ssl.verification_mode: certificate
xpack.security.http.ssl.key: /etc/elasticsearch/certs/es-node-1/es-node-1.key     
xpack.security.http.ssl.certificate: /etc/elasticsearch/certs/es-node-1/es-node-1.crt     
xpack.security.http.ssl.certificate_authorities: [ "/etc/elasticsearch/certs/ca/ca.crt" ] 
```

> Node 02 (transport layer + http layer)

```yml
# /etc/elasticsearch/elasticsearch.yml 
xpack.security.transport.ssl.enabled: true
xpack.security.transport.ssl.verification_mode: certificate
xpack.security.transport.ssl.key: /etc/elasticsearch/certs/es-node-2/es-node-2.key
xpack.security.transport.ssl.certificate: /etc/elasticsearch/certs/es-node-2/es-node-2.crt
xpack.security.transport.ssl.certificate_authorities: [ "/etc/elasticsearch/certs/ca/ca.crt" ]                

xpack.security.http.ssl.enabled: true
xpack.security.http.ssl.verification_mode: certificate
xpack.security.http.ssl.key: /etc/elasticsearch/certs/es-node-2/es-node-2.key     
xpack.security.http.ssl.certificate: /etc/elasticsearch/certs/es-node-2/es-node-2.crt     
xpack.security.http.ssl.certificate_authorities: [ "/etc/elasticsearch/certs/ca/ca.crt" ] 
```

6) SSL settings for Filebeat:

```yml
# /etc/filebeat/filebeat.yml
output.elasticsearch.hosts: ['172.16.1.2:9200','172.16.1.3:9200']
output.elasticsearch.protocol: https
output.elasticsearch.ssl.certificate: "/etc/filebeat/certs/es-node-1/es-node-1.crt"
output.elasticsearch.ssl.key: "/etc/filebeat/certs/es-node-1/es-node-1.key"
output.elasticsearch.ssl.certificate_authorities: ["/etc/filebeat/certs/ca/ca.crt"]
```

7) SSL settings for Kibana:

> Kibana <-> Browser 

```yml
# /etc/kibana/kibana.yml
server.ssl.enabled: true
server.ssl.certificate: "/etc/kibana/certs/es-node-1/es-node-1.crt"
server.ssl.key: "/etc/kibana/certs/es-node-1/es-node-1.key"
```

> Kibana <-> Elasticsearch

```yml
# /etc/kibana/kibana.yml
elasticsearch.hosts: ["https://172.16.1.2:9200"]
elasticsearch.ssl.certificateAuthorities: ["/etc/kibana/certs/ca/ca.crt"]
elasticsearch.ssl.certificate: "/etc/kibana/certs/es-node-1/es-node-1.crt"
elasticsearch.ssl.key: "/etc/kibana/certs/es-node-1/es-node-1.key"
```

All is working as expected, also, Filebeat command for testing the output is now even better because we are pointing it directly to Elasticsearch:

```
filebeat test output
elasticsearch: https://172.16.1.2:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 172.16.1.2
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.2
    dial up... OK
  talk to server... OK
  version: 7.0.0
elasticsearch: https://172.16.1.3:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 172.16.1.3
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.2
    dial up... OK
  talk to server... OK
  version: 7.0.0
```

![image](https://user-images.githubusercontent.com/8860227/57542296-85739300-7351-11e9-907e-e2718c5bfab6.png)

**Comment by jesusgn90** [comment]
## Update 13/05/2019

Today's work was focused mainly on our documentation. 

Here are the additions and changes of today for our docs:

- https://github.com/wazuh/wazuh-documentation/commit/56c8cc11d6a59ad29f5d3e6d7406149065944249 adapt the Search Guard guide for 7.0.1 and removing Logstash from the guide, now it's using Filebeat.
- https://github.com/wazuh/wazuh-documentation/commit/b5219b00f1474a70154b02ef11bda255e7663265 new section named _Protect your data in the Elastic Stack_ where I explain X-Pack, Search Guard and some other docs as the NGINX reverse proxy for Kibana, now all is placed in the same guide and it has been reduced and simplified for the user.
  - The X-Pack guide now covers all steps described here https://github.com/wazuh/wazuh-kibana-app/issues/1433#issuecomment-491350214 in a clear and easy guide.
![image](https://user-images.githubusercontent.com/8860227/57637225-290bb000-75ab-11e9-950c-a6c600fd7250.png)
 
- The branch _elastic-7_ has been updated with the latest changes from _3.9_ https://github.com/wazuh/wazuh-documentation/commit/3e31106a9547704b1f4d7519269bd971f0042d09
- https://github.com/wazuh/wazuh-documentation/commit/9c02d4aa04f0f814380adf6faf49ebfa3fbe5ee4 removed all the docs inside the Wazuh app user manual regarding X-Pack, now we have a clear section about X-Pack.

With the Elastic 7 rework, some other opened tickets will be solved such as https://github.com/wazuh/wazuh-documentation/issues/997 or https://github.com/wazuh/wazuh-documentation/issues/1194 for example.

**Comment by juankaromo** [comment]
**Update 15/05**

I have continued to change some containers, tables, and other elements we have in the app for `eui` elements, so that they adapt natively to the dark theme. e.g:

![image](https://user-images.githubusercontent.com/21195730/57784051-87619b80-772f-11e9-87c4-5fb4996c43cc.png)

**Comment by jesusgn90** [comment]
## Update 15/05/2019

Today I've been mainly focused on our documentation, there is still some needed changes.

https://github.com/wazuh/wazuh-documentation/commit/2a634ab74fd01771436c3ca368b14f92d1030aea 
> Missing Elastic repository for installing Filebeat.

https://github.com/wazuh/wazuh-documentation/commit/8e0c8c315e94ac6df5818e783a707c562db77266
> Replace most of the app screenshots we have across the documentation, some of them were very old and our GUI has changed. Other screenshots were just wrong due to the product changes, now they are consistent with our latest changes. This task is still in progress.

https://github.com/wazuh/wazuh-documentation/commit/faa7764d06f8551d34894d488018b9655f8f1dd6
> Changes related to "user-manual/kibana-app/*" sections. This commit simplifies all those documents (some of them were so much redundant or extremely complex for the user). Other things we were saying were so much obvious so I decided to simply remove them. In addition, the app has changed a lot and some of those documents were outdated/incomplete so I fixed them adding the new changes and correcting the wrong ones.

**Comment by jesusgn90** [comment]
## Update 16/05/2019

**Documentation**

https://github.com/wazuh/wazuh-documentation/commit/575f6190bab060decea0960045e8a6532d344713

- The "Protecting your data" section has been moved inside the "Installing Elastic Stack"
- "Without sudo" way to install the app has been removed.
- Added removing "@src_ip" field in the ingest processor, **pending testing**
- Removed the warning in the "Upgrading Elastic Stack" docs.
- Split  "Upgrading Elastic Stack" docs into three new docs: "From 6.x to 6.7", "From 6.7 to 7.x" and "From 7.x to 7.y"
- Corrected some steps from  "Upgrading Elastic Stack" docs after testing the guide.
- Added an extra step for "update_by_query" and its "timestamp" related curl command.
- Unified  "Upgrading Elastic Stack" different docs for DEB-based and RPM-based distributions, there is no need to having two different guides just for the package manager.
- Added a new header for clarifying why we removed Logstash (it's available optionally if needed), also explain why you can remove Java too.

**Wazuh app**

https://github.com/wazuh/wazuh-kibana-app/commit/ce247d08cf8ac8acd7e0b7f49fdff6f8ea60734a

> Force `timestamp` field to be the new `timeFieldName` in index patterns for alerts.

https://github.com/wazuh/wazuh-kibana-app/commit/9bcfa1ba2dcde61ad334f9810ca2ac69fa77ddce

> Force `timestamp` field to be the new `timeFieldName` in index patterns for monitoring events.

All is working as expected after an upgrade from 6.7.2 to 7.0.1

**Comment by jesusgn90** [comment]
## 17/05/2019

Documentation:

- https://github.com/wazuh/wazuh-documentation/commit/14fb0c0c493da3b142391e8869e4990fe25c7b71 added `ignore_missing` for the GeoIP processor, otherwise, it fails if the field is missing.
- https://github.com/wazuh/wazuh-documentation/commit/a46e3c8c626f3e4ca4bfe14563ab5b1e9931cddd replaced more old screenshot with some other ones more up-to-date.
- https://github.com/wazuh/wazuh-documentation/commit/2c564a5098d51e911437480e56b0ff5c484c74e6 reviewed and adapted the "Elastic tuning" docs for Elastic 7
- https://github.com/wazuh/wazuh-documentation/commit/97a34daf68606dac43bfaecae7e71e3789864c35 small typo in the automatic API command
- https://github.com/wazuh/wazuh-documentation/commit/957aeb97cf973d0e4d5b3b432fbb000e341ce57e correct config files path due to https://github.com/wazuh/wazuh/commit/2440bf4ed0d6c199cd4e64bbb33c1ebbe7b96846
- https://github.com/wazuh/wazuh-documentation/commit/48c7cc411a49f04f7a69d23dd0130e480e23be59 added a new section named "Transform your data with Logstash" for those users that may need Logstash for event enrichment/filtering

Configuration files:

- https://github.com/wazuh/wazuh/commit/2440bf4ed0d6c199cd4e64bbb33c1ebbe7b96846 new directory tree.

```
extensions/
├── elasticsearch/
│   ├── restore_alerts/
│   │   ├── README.md
│   │   ├── restore_alerts.conf
│   │   └── restore_alerts.sh
│   ├── 6.x/
│   │   └── wazuh-template.json
│   ├── 7.x/
│       └── wazuh-template.json
├── filebeat/
│   ├── 6.x/
│   │   └── filebeat.yml
│   ├── 7.x/
│       └── filebeat.yml
├── logstash/
│   └── 6.x/
│       ├── 01-wazuh-local.conf
│       └── 01-wazuh-remote.conf
│   └── 7.x/
│       └── 01-wazuh-remote.conf
```
